pkg/retester/retester_test.go

```diff
@@ -567,3 +580,195 @@ pr3:
+		expectedString string
+	}{
+		{
+			name:  "hold PR",
+			cache: backoffCache{cache: map[string]*pullRequest{"org/repo#123": {PRSha: "holdPR", RetestsForBaseSha: 3, RetestsForPrSha: 9}}, logger: logger},
+			pr: tide.PullRequest{Number: githubv4.Int(123),
+				Repository: struct {
+					Name          githubv4.String
+					NameWithOwner githubv4.String
+					Owner         struct{ Login githubv4.String }
+				}{Name: "repo", NameWithOwner: "org/repo", Owner: struct{ Login githubv4.String }{Login: "org"}},
+				HeadRefOID: "holdPR"},
+			policy:         RetesterPolicy{3, 9, &True},
+			expected:       0,
+			expectedString: "Revision holdPR was retested 9 times: holding",
+		},
+		{
+			name:  "pause PR",
+			cache: backoffCache{cache: map[string]*pullRequest{"org/repo#123": {PRSha: "pausePR", RetestsForBaseSha: 3, RetestsForPrSha: 3}}, logger: logger},
+			pr: tide.PullRequest{Number: githubv4.Int(123),
+				Repository: struct {
+					Name          githubv4.String
+					NameWithOwner githubv4.String
+					Owner         struct{ Login githubv4.String }
+				}{Name: "repo", NameWithOwner: "org/repo", Owner: struct{ Login githubv4.String }{Login: "org"}},
+				HeadRefOID: "pausePR"},
+			policy:         RetesterPolicy{3, 9, &True},
+			expected:       1,
+			expectedString: "Revision pausePR was retested 3 times against base HEAD : pausing",
+		},
+		{
+			name:           "retest PR",
+			cache:          backoffCache{cache: map[string]*pullRequest{}, logger: logger},
+			pr:             tide.PullRequest{HeadRefOID: "retestPR"},
+			policy:         RetesterPolicy{3, 9, &True},
+			expected:       2,
+			expectedString: "Remaining retests: 2 against base HEAD  and 8 for PR HEAD retestPR in total",
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			actual, actualString := tc.cache.check(tc.pr, tc.baseSha, tc.policy)
+			if diff := cmp.Diff(tc.expected, actual); diff != "" {
+				t.Errorf("%s differs from expected:\n%s", tc.name, diff)
+			}
+			if diff := cmp.Diff(tc.expectedString, actualString); diff != "" {
+				t.Errorf("%s differs from expected:\n%s", tc.name, diff)
+			}
+		})
+	}
+}
+
+var (
+	sevenDays             = 24 * 7 * time.Hour
+	emptyRetestController = &RetestController{}
+)
+
+func TestNewController(t *testing.T) {
+	testCases := []struct {
+		name           string
+		ghClient       githubClient
+		cfg            config.Getter
+		gitClient      git.ClientFactory
+		usesApp        bool
+		cacheFile      string
+		cacheRecordAge time.Duration
+		config         *Config
+		expected       *RetestController
+	}{
+		{
+			name:           "basic",
+			cacheFile:      "basic_case.yaml",
+			cacheRecordAge: sevenDays,
+			expected:       &RetestController{backoff: &backoffCache{file: "testdata/NewController/basic_case.yaml", cacheRecordAge: sevenDays}},
+		},
+		{
+			name:           "empty",
+			cacheFile:      "",
+			cacheRecordAge: sevenDays,
+			expected:       emptyRetestController,
+		},
+	}
```

Comment:
I am trying to understand the difference between the 2 cases:
Can you show me a line of code for each case that is not covered by the other?